paddle/fluid/operators/fc_op.cc

```diff
@@ -110,6 +119,25 @@ void FCOpMaker::Make() {
   AddInput("W", "(Tensor), The weight fc op with shape (I, O).");
   AddInput("Bias", "(Tensor, optional) Bias vector with shape (1 x O")
       .AsDispensable();
+
+  AddAttr<float>("Scale_in",
+                 "(float, default 1.0f), The quantize scale of input data")
+      .SetDefault(1.0f);
+  AddAttr<std::vector<float>>("Scale_weights",
```

Comment:
Channel-wise quantization should only be applied on the convolution op. There is no channel concept for fc/mul op. So, I think the weight of fc/mul  op should do the whole tensor quantization, namely one scale value. TensoRT also takes this strategy.

Reply:
We supporting both tensor wise and channel wise weights scale in kernel, channel is dim-1 exactly. Accuracy with channel wise scale will be better.

Reply:
In Paddle, the data layout of the fc/mul weight is input_node_num * output_node_num.  You said "the channel is dim-1 exactly",  what does it mean `output_node_num` ?

Reply:
> In Paddle, the data layout of the fc/mul weight is input_node_num * output_node_num. You said "the channel is dim-1 exactly", what does it mean `output_node_num` ?

Yes, it means output.

> > > If weights transpose:
> > > 
> > > * Is the `cpu_int8_save_model` different with other `int8_save_model`? And you should do another weights transpose before `save_model`?
> > > * How do you deal with `fc` + `fc`?
> > 
> > 
> > 
> > 1. It depends on how the API defines. E.g., if int8_save_model aims to provide device-agnostic int8 model (for both GPU and CPU), you can either save plain weight or transposed weight (but need to let device know with clear API definition).
> > 2. It does not matter. We verified in AlexNet, GoogleNet, and VGGxx.
> 
> 1. I think you should save the int8 weight as origin data layout, not the transposed data layout. 
When you load saved weights, do the transpose operation in the pass at first.

Agree, that's a better choice.

> 2. In Paddle, data layout described as follow: A = batch_size * input_node_num, W = input_node_num * output_node_num,
>    O = A * W = batch_size * output_node_num.
>    If weights transpose:
>    W should be transposed as WT = output_node_num *  input_node_num, A should be transposed as
>    AT = input_node_num *  batch_size. Then:
>    OT = WT * AT = output_node_num * batch_size.
>    In conclusion, is OT your output's data layout? If so, I think you should transpose OT to get O(data layout).

Thanks, I understood your point here. My point is that MKL-DNN supports FC INT8 with channel-wise quantization scales per output channel. In order to enable channel-wise for better INT8 accuracy, frameworks or ops can decide whether to transpose based on the API definition of saving weight format.

Reply:
> In order to enable channel-wise for better INT8 accuracy

This better INT8 accuracy is on CV models or all other models? Do you have detail benchmark before and after enable channel-wise?